src/test/ui/issue-18400.stderr

```diff
@@ -0,0 +1,75 @@
+error[E0275]: overflow evaluating the requirement `_: std::marker::Sized`
+  --> $DIR/issue-18400.rs:34:7
+   |
+LL |     0.contains(bits);
+   |       ^^^^^^^^
+   |
+   = help: consider adding a `#![recursion_limit="128"]` attribute to your crate
+   = note: required because of the requirements on the impl of `Set<&[_]>` for `{integer}`
+   = note: required because of the requirements on the impl of `Set<&[&[_]]>` for `{integer}`
```

Comment:
Probably doing the same thing we do in other cases involving recursion (like #49950): detect the smallest non repeating sequence and only present that. In this case we could present the first 3 and the last 2 and it would be just as useful (but I could think of cases where the sequence needs more than the first three cycles to show itself as repeating).